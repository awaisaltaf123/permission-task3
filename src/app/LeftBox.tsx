import React from "react";
import Image from "next/image";

export const Spinner: React.FC = () => {
  return (
    <div
      data-testid="spinner"
      className="spinner w-6 h-6 rounded-full animate-spin
    border-4 border-solid border-white-500 border-t-transparent pr-2"
    ></div>
  );
};

interface LeftBoxProps {
  guid: string;
  isLoading: boolean;
  isValidGuid: boolean | null;
  onGuidChange: (guid: string) => void;
  onSubmit: any;
}

const LeftBox: React.FC<LeftBoxProps> = ({
  guid,
  isLoading,
  isValidGuid,
  onGuidChange,
  onSubmit,
}) => {
  return (
    <div className="form-box flex-1 ml-4" data-testid="left-box">
      <form
        onSubmit={onSubmit}
        className="box form p-4 bg-white border border-black rounded p-4 my-4 flex flex-col"
      >
        <div className="flex justify-between">
          <label className="label">Enter a valid GUID here.</label>
          {isValidGuid === false && (
            <span className="error-message text-red-500 bg-red-100 px-4 py-2 rounded-full">
              Invalid GUID
            </span>
          )}
        </div>
        <input
          type="text"
          value={guid}
          onChange={(e) => onGuidChange(e.target.value)}
          disabled={isLoading}
          className="input border border-black rounded-full p-2 my-4"
          placeholder="Enter GUID"
        />
        <button
          type="submit"
          disabled={isLoading}
          className="button bg-yellow-400 rounded-full flex items-center justify-center w-44 h-10"
        >
          {isValidGuid && (
            <Image src="/tick.png" alt="Valid" width={24} height={24} />
          )}
          {isLoading && <Spinner />}
          {"Submit"}
        </button>
      </form>
      <div className="text-box border border-black rounded p-4 my-4 bg-white w-1/4 break-words">
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry's standard dummy text ever
        since the 1500s.
      </div>
      <style jsx>{`
        @media (max-width: 800px) {
          .text-box {
            width: auto;
          }
          .form-box {
            margin-right: 15px;
          }
        }
      `}</style>
    </div>
  );
};

export default LeftBox;
