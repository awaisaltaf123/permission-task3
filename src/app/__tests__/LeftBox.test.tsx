import React from "react";
import { render, fireEvent } from "@testing-library/react";
import LeftBox, { Spinner } from "../LeftBox";
import "@testing-library/jest-dom";

describe("LeftBox", () => {
  it("renders the input field and submit button", () => {
    const { container } = render(
      <LeftBox
        guid={""}
        isLoading={false}
        isValidGuid={null}
        onGuidChange={function (guid: string): void {
          throw new Error("Function not implemented.");
        }}
        onSubmit={undefined}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls onGuidChange when input value changes", () => {
    const onGuidChange = jest.fn();
    const { getByPlaceholderText } = render(
      <LeftBox
        guid=""
        onGuidChange={onGuidChange}
        isLoading={false}
        isValidGuid={null}
        onSubmit={undefined}
      />
    );

    const inputElement = getByPlaceholderText("Enter GUID");
    fireEvent.change(inputElement, { target: { value: "123456" } });

    expect(onGuidChange).toHaveBeenCalledWith("123456");
  });

  it("calls onSubmit when submit button is clicked", () => {
    const onSubmit = jest.fn();
    const { getByText } = render(
      <LeftBox
        onSubmit={onSubmit}
        guid={""}
        isLoading={false}
        isValidGuid={null}
        onGuidChange={function (guid: string): void {
          throw new Error("Function not implemented.");
        }}
      />
    );

    const submitButton = getByText("Submit");
    fireEvent.click(submitButton);

    expect(onSubmit).toHaveBeenCalled();
  });

  it("renders error message when isValidGuid is false", () => {
    const { getByText } = render(
      <LeftBox
        isValidGuid={false}
        guid={""}
        isLoading={false}
        onGuidChange={function (guid: string): void {
          throw new Error("Function not implemented.");
        }}
        onSubmit={undefined}
      />
    );
    const errorMessage = getByText("Invalid GUID");
    expect(errorMessage).toBeInTheDocument();
  });

  it("renders valid image when isValidGuid is true", () => {
    const { getByAltText } = render(
      <LeftBox
        isValidGuid={true}
        guid={""}
        isLoading={false}
        onGuidChange={function (guid: string): void {
          throw new Error("Function not implemented.");
        }}
        onSubmit={undefined}
      />
    );
    const validImage = getByAltText("Valid");
    expect(validImage).toBeInTheDocument();
  });

  it("renders spinner when isLoading is true", () => {
    const { getByTestId } = render(
      <LeftBox
        isLoading={true}
        guid={""}
        isValidGuid={null}
        onGuidChange={function (guid: string): void {
          throw new Error("Function not implemented.");
        }}
        onSubmit={undefined}
      />
    );
    const spinner = getByTestId("spinner");
    expect(spinner).toBeInTheDocument();
  });

  it("renders spinner correctly", () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
