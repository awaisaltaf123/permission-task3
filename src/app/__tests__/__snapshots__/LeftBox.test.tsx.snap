// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftBox renders spinner correctly 1`] = `
<div>
  <div
    class="spinner w-6 h-6 rounded-full animate-spin
 border-4 border-solid border-white-500 border-t-transparent pr-2"
    data-testid="spinner"
  />
</div>
`;

exports[`LeftBox renders the input field and submit button 1`] = `
<div>
  <div
    class="jsx-1788078474 form-box flex-1 ml-4"
    data-testid="left-box"
  >
    <form
      class="jsx-1788078474 box form p-4 bg-white border border-black rounded p-4 my-4 flex flex-col"
    >
      <div
        class="jsx-1788078474 flex justify-between"
      >
        <label
          class="jsx-1788078474 label"
        >
          Enter a valid GUID here.
        </label>
      </div>
      <input
        class="jsx-1788078474 input border border-black rounded-full p-2 my-4"
        placeholder="Enter GUID"
        type="text"
        value=""
      />
      <button
        class="jsx-1788078474 button bg-yellow-400 rounded-full flex items-center justify-center w-44 h-10"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      class="jsx-1788078474 text-box border border-black rounded p-4 my-4 bg-white w-1/4 break-words"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.
    </div>
  </div>
</div>
`;
